src/SmartComponents/Inventory/applications/compliance/SystemRulesTable.js

```diff
@@ -1,181 +1,362 @@
+                this.setState(() => ({
+                    currentRows,
+                    originalRows: rows,
+                    rows: onlyPassedRows
+                }));
+            });
+        } else {
+            this.currentRows(this.state.page, this.state.itemsPerPage, this.state.originalRows).then((currentRows) => {
+                this.setState(() => ({
+                    currentRows,
+                    rows: this.state.originalRows
+                }));
+            });
+        }
     }
 
     render() {
-        const currentRows = this.currentRows();
+        const { sortBy, rows, currentRows, columns, page, itemsPerPage } = this.state;
         const { loading } = this.props;
-        return (
-            <React.Fragment>
-                <Grid gutter="sm">
-                    <GridItem span={ 10 }>
-                        <Input
-                            id="search"
-                            type="text"
-                            style={ { width: '200px' } }
-                        />{ ' ' }
-                        <SearchIcon style={ { paddingTop: '4px' } } />
-                    </GridItem>
-                    <GridItem span={ 2 }>
-                        { !loading && <ComplianceRemediationButton selectedRules={ this.selectedRules() } /> }
-                    </GridItem>
-
-                    <GridItem span={ 12 }>
-                        <Table
-                            variant="large"
-                            header={ [ 'Rule', 'Reference ID', 'Policy', 'Severity', 'Passed' ] }
-                            hasCheckbox
-                            expandable
-                            onItemSelect={ this.onItemSelect }
-                            rows={ loading ? [ ...Array(10) ].map(() => ({
-                                cells: [{
-                                    title: <RowLoader />,
-                                    colSpan: 5
-                                }]
-                            })) : currentRows.map((oneRow, key) => ({
-                                ...oneRow,
-                                ...oneRow.hasOwnProperty('isOpen') ?
-                                    { isOpen: this.state.openNodes.indexOf(key) !== -1 } :
-                                    {}
-                            })
-                            ) }
-                            onExpandClick={ this.onExpandClick }
-                            footer={ <Pagination
-                                numberOfItems={ (this.state.rows.length - 2) / 2 }
-                                onPerPageSelect={ this.setPerPage }
-                                page={ this.state.page }
-                                onSetPage={ this.setPage }
-                                itemsPerPage={ this.state.itemsPerPage }
-                            /> }
-                        />
-                    </GridItem>
-                </Grid>
-            </React.Fragment>
-        );
+        if (loading) {
```

Comment:
I think there's a problem here: I've seen that every now and then, when the component finally renders with the right props (`profileRules` is populated from GraphQL), the table *does not* render again with the right rows. The user can workaround this by clicking on next page to trigger an event that fills currentRows and renders the page again.. but I suppose there's some way I can force rerender once the promise is finished

Reply:
Fixed now by using 'setState' on the constructor to force the reload once the currentRows promise is fulfilled.